hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/MySQL/FederationStateStoreStoredProcs.sql

```diff
@@ -122,10 +122,26 @@ BEGIN
    WHERE applicationId = applicationID_IN;
 END //
 
-CREATE PROCEDURE sp_getApplicationsHomeSubCluster()
-BEGIN
-   SELECT applicationId, homeSubCluster
-   FROM applicationsHomeSubCluster;
+CREATE PROCEDURE sp_getApplicationsHomeSubCluster(IN limit_IN int, IN homeSubCluster_IN varchar(256))
+BEGIN
+   SELECT
+       applicationId,
+       homeSubCluster,
+       createTime
+   FROM
+    (SELECT
+       applicationId,
+       homeSubCluster,
+       createTime,
+       @app_rank := IF(@home_sc = homeSubCluster, @app_rank + 1, 1) AS app_rank,
+       @home_sc := homeSubCluster
+  	 FROM applicationshomesubcluster
```

Comment:
The tabs in line warnings are legit. Let's replace them with spaces.

Reply:
Thanks for your help reviewing the code, I will fix it.